docs/contributor/02-10-testing-strategy.md

```diff
@@ -0,0 +1,65 @@
+Therefore, we need to decide on a testing strategy that balances the need for comprehensive
+testing with the requirement for fast feedback and development.
+
+## Testing strategy outline
+
+The testing strategy for the Istio module will be implemented according to the following guidelines:
+1. Tests that depend on the Gardener platform will not run on Pull Requests (PRs).
+2. Gardener related tests will run during post-merge workflows and on scheduled runs.
+3. If a Gardener-related test fails, the cluster will remain alive for debugging purposes.
+4. In the event of a test failure during a post-merge workflow, the PR owner is responsible for resolving the issue.
+5. PR tests should generally avoid relying on external resources.
+   This especially means that secrets should not be required for running PR tests whenever possible.
+6. Integration tests for PRs will run on a local Kubernetes cluster using the k3d platform.
+7. Compatibility, performance, and UI tests will run only on scheduled runs.
+8. Tests ensuring release stability and readiness will be triggered during the release workflow.
+
+Additionally, the following naming conventions will be adopted for workflows:
+- Workflows that run before merge should be prefixed with `pull`.
+- Workflows triggered after merge should be prefixed with `post`.
+- Workflows running on schedule should be prefixed with `schedule`.
+- Workflows related to release should be prefixed with `release`.
+- Workflows that run on manual trigger will be prefixed with `call`.
+
+## Rationale
+
+The primary objective is to ensure that the module remains stable and prepared for release,
+while providing prompt feedback for PRs.
+This testing strategy prioritizes comprehensive e2e evaluations during post-merge and release pipeline stages,
+while PR-level assessments focus on unit tests and fundamental integration tests.
+Testing on Gardener clusters incurs significant computational and time costs.
+End-to-end (e2e) Kubernetes scenarios are instead evaluated using k3d where feasible,
+due to its reduced initialization time and higher reliability in controlled environments.
+The probability of Gardener-specific test failures when k3d-based tests pass is considered low.
+In the event of Gardener-related test failures after a PR has been merged,
+the responsibility for stabilizing the associated workflows lies with the author of the merged PR.
+
+## Separation of concerns
+
+To achieve the desired testing strategy,
+the tests will be separated into three categories,
+that are defined as follows:
+- `unit tests` that run without any external dependencies, and do not require any pre-existing resources.
+- `integration tests` that require external dependencies,
+but do not require a full Kubernetes cluster (for example, they might run on controller-runtime `envtest` environment).
+- `e2e tests` that require a full Kubernetes cluster.
```

Comment:
I would even split it into:
- lightweight e2e tests with lightweight kubernetes (k3d) - everything runs in Docker in a local machine, there is no hyperscaler involved yet (no real networking, no load balancer, etc.)
- full e2e tests on a real infrastructure

Reply:
The above 'pyramid' says about regular/usual functional tests, but I think we should also mention 'special' cases, like:
- UI tests
- performance tests
- upgrade tests
- zero downtime tests
etc.